xdrgen needs to be resynced with nightly

Fails to build:

```
rust-xdr$ cargo test --features unstable
   Compiling regex-syntax v0.3.9
   Compiling log v0.3.6
   Compiling libc v0.2.19
   Compiling nom v1.2.4
   Compiling winapi v0.2.8
   Compiling utf8-ranges v0.1.3
   Compiling memchr v0.1.11
   Compiling winapi-build v0.1.1
   Compiling aho-corasick v0.5.3
   Compiling kernel32-sys v0.2.2
   Compiling thread-id v2.0.0
   Compiling thread_local v0.2.7
   Compiling regex v0.1.80
   Compiling env_logger v0.3.5
   Compiling xdrgen v0.2.4 (file:///home/imp/dev/rust/crates/rust-xdr/xdrgen)
error[E0432]: unresolved import `syntax::parse::token::str_to_ident`
 --> xdrgen/src/spec/rustast.rs:9:9
  |
9 | pub use syntax::parse::token::str_to_ident;
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `str_to_ident` in `parse::token`

error[E0425]: cannot find function `intern` in module `syntax::parse::token`
  --> xdrgen/src/spec/fake_extctxt.rs:18:42
   |
18 |       format: syntax::codemap::MacroBang(syntax::parse::token::intern("")),
   |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `syntax::parse::token`

error[E0061]: this function takes 3 parameters but 4 parameters were supplied
  --> xdrgen/src/spec/rustast.rs:21:56
   |
21 | 	let mut p = syntax::parse::new_parser_from_source_str(&ctxt.parse_sess, Vec::new(), "<peg>".to_string(), e.to_string());
   | 	                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 parameters

error[E0061]: this function takes 3 parameters but 4 parameters were supplied
  --> xdrgen/src/spec/rustast.rs:32:56
   |
32 | 	let mut p = syntax::parse::new_parser_from_source_str(&ctxt.parse_sess, Vec::new(), "<peg>".to_string(), e.to_string());
   | 	                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 parameters

error[E0061]: this function takes 3 parameters but 4 parameters were supplied
  --> xdrgen/src/spec/rustast.rs:39:56
   |
39 | 	let mut p = syntax::parse::new_parser_from_source_str(&ctxt.parse_sess, Vec::new(), "<peg>".to_string(), e.to_string());
   | 	                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 parameters

error[E0061]: this function takes 3 parameters but 4 parameters were supplied
  --> xdrgen/src/spec/fake_extctxt.rs:10:48
   |
10 |   let mut cx = syntax::ext::base::ExtCtxt::new(&ps, Vec::new(),
   |                                                ^^^^^^^^^^^^^^^ expected 3 parameters

error: aborting due to 4 previous errors

error: Could not compile `xdrgen`.

To learn more, run the command again with --verbose.
```

--- a/xdrgen/tests/lib.rs
+++ b/xdrgen/tests/lib.rs
@@ -45,12 +45,12 @@ fn main() {{}}
 
     let compile = {
 	    let mut cmd = Command::new("rustc");
-	    let cmd = cmd
+        let cmd = cmd
                .current_dir(std::env::current_dir()?)
                .arg("--crate-type").arg("bin")
                .arg("--crate-name").arg(name)
-		       .arg("-L").arg("./target/debug/deps")
-		       .arg("--extern").arg("xdr_codec=target/debug/deps/libxdr_codec.rlib")
+		       .arg("-L").arg("../target/debug/deps")
+		       .arg("--extern").arg("xdr_codec=../target/debug/libxdr_codec.rlib")
 		       .arg("-o").arg(exefile)
 		       //.arg("-Z").arg("no-trans")
 		       .arg(mainfile);

